### PR TITLE
fix: chunkIntoN to chunk correctly

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Fix the chunk size helper. 

**Description**

- The function is used handling the batch request handling 
- Earlier the chunkSize helper was using the wrong chunk size 

**Steps to test or reproduce**

Run all tests

### CODE REVIEW

File: packages/prover/test/unit/utils/conversion.test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        output: [
+          [6, 5],
+          [4, 3],
+          [2, 1],
+        ],
+      },
+    ];
+
+    for (const {title, input, output} of testCases) {
+      it(`should split the chunks correctly for "${title}"`, async () => {
```
Comment (nflaig) [inline packages/prover/test/unit/utils/conversion.test.ts]: What about just printing out the title, this seems repetitive to me

File: packages/prover/test/unit/utils/conversion.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      {
+        title: "data 1 extra than chunk size",
+        input: {
+          data: [1, 2, 3, 4],
+          n: 3,
+        },
+        output: [[1, 2, 3], [4]],
+      },
+      {
+        title: "when data have different order",
```
Comment (nflaig) [inline packages/prover/test/unit/utils/conversion.test.ts]: Does this test case make sense? it makes it seem like the type of elements are relevant, rather would add 1-2 tests cases that show that it works with any type of elements, e.g. objects or nested arrays

Comment (nazarhussain) [reply packages/prover/test/unit/utils/conversion.test.ts]: Yes just want to make sure doing chunk does not change the order.

Comment (nflaig) [reply packages/prover/test/unit/utils/conversion.test.ts]: All the other existing test cases already assert that. 

My point is that why would the values of elements impact that?

> data have different order

Makes it sounds like this `chunkIntoN` inspects the value of elements, which it does not. IMO better way to prove this is to use different types of elements.

File: packages/prover/test/unit/utils/conversion.test.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import {expect} from "chai";
+import {chunkIntoN} from "../../../src/utils/conversion.js";
+
+describe("utils/conversion", () => {
+  describe("chunkIntoN", () => {
+    const testCases = [
+      {
+        title: "when even number of chunks",
```
Comment (nflaig) [inline packages/prover/test/unit/utils/conversion.test.ts]: what about just

```suggestion
        title: "even number of chunks",
```
You can keep your preferred phrasing, just wondering what would be the best test descriptions for such cases in general in terms of readability

**Comment by nazarhussain** [comment]
@nflaig When reading the test we should read in full context from the description. So I believe current pattern feels more natural to read.

**Review by nflaig — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.12.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.12.0) :tada:

**Pull request merged**